src/server/api/lib/twilio.js

```diff
@@ -702,6 +702,44 @@ async function addNumbersToMessagingService(
   );
 }
 
+/**
+ * Release a phone number and delete it from the owned_phone_number table
+ */
+async function deleteNumber(twilioInstance, phoneSid, phoneNumber) {
+  const response = await twilioInstance.incomingPhoneNumbers(phoneSid).remove();
+  if (response.error) {
```

Comment:
I think my one request here is that instead of throwing if deleting a number fails, we should just count/track the errors here.  A use case that came up for us and at least one other campaign was deleting numbers in Spoke when at least some of them were already deleted from the Twilio console.

Other than that, looks good.